Lib/test/test_memoryview.py

```diff
@@ -71,6 +71,36 @@ def setitem(value):
         m = None
         self.assertEqual(sys.getrefcount(b), oldrefcount)
 
+    def test_setitem_readonly_and_writable_buffer(self):
+        if not self.rw_type:
+            self.skipTest("no writable type to test")
+        b = self.rw_type(self._source)
+        oldrefcount = sys.getrefcount(b)
+        m = self._view(b)
+        readonly_m = m.cast("B", readonly=True)
+        def setitem(value):
+            readonly_m[0] = value
+        self.assertRaises(TypeError, setitem, 12)
```

Comment:
Also check that `readonly_m.readonly` is True?

Same for the other test below.